webknossos/webknossos/dataset/_metadata.py

```diff
@@ -0,0 +1,104 @@
+from contextlib import contextmanager
+from typing import (
+    Any,
+    Dict,
+    Generator,
+    Iterator,
+    List,
+    Sequence,
+    TypeVar,
+    Union,
+)
+
+from webknossos.client.api_client.models import ApiDataset, ApiFolder, ApiMetadata
+from webknossos.utils import infer_metadata_type, parse_metadata_value
+
+_T = TypeVar("_T", bound="Metadata")
+
+
+class Metadata(dict):
```

Comment:
Have you considered implementing a `MutableMapping` directly? I think that would remove some trouble with the persistence that a `dict` does automatically.

```suggestion
class Metadata(MutableMapping):
```